build.gradle

```diff
@@ -264,6 +267,11 @@ dependencies {
     gradlePlugins project(':workers')
     gradlePlugins project(':dependencyManagement')
     gradlePlugins project(':testKit')
+
+    coreRuntimeExtensions project(':dependencyManagement') //See: DynamicModulesClassPathProvider.GRADLE_EXTENSION_MODULES
+    coreRuntimeExtensions project(':pluginUse')
+    coreRuntimeExtensions project(':workers')
+    coreRuntimeExtensions patchedExternalModules
```

Comment:
This is not very nice... 

I find it hard to grasp how the _core runtime_ is defined in general as it is partially defined here but also influences by DynamicModulesClassPathProvider.  Improving this feels out of scope for this issue.